trinity/protocol/common/events.py

```diff
@@ -0,0 +1,11 @@
+from p2p.peer import PeerMessage
+
+from trinity.extensibility.events import BaseEvent
+
+
+class NewChainTipEvent(BaseEvent):
+    """
+    Broadcasted when a new tip (in regular or light protocol) is received from a peer.
+    """
+    def __init__(self, peer_message: PeerMessage) -> None:
+        self.peer_message = peer_message
```

Comment:
This seems like it would be good to transmit this data in a normalized fashion rather than the raw message (which is what I assume this is doing).

Reply:
It's a named tuple with some important bits split out https://github.com/ethereum/py-evm/blob/fe2019b8af7253810328181be50d9638b98f0edf/p2p/peer.py#L616-L619

So it's going to have the same pickleable problem as the others.

Reply:
Yeah, I was actually thinking that it would be a custom payload which contained maybe a peer identifier and then the normalized tip info (`head_td`, `head_number`, `head_hash`?)